pioneers-testnet/pioneers-exercise-2.md

```diff
@@ -0,0 +1,170 @@
+
+4. Generate a signing key, txsign.
+
+```bash
+$ cardano-cli keygen …
+```
+
+Sign your transaction in txbody using the signing key for *myaddr*:
+
+```bash
+$ cardano-cli shelley transaction sign \
+ 	--tx-body-file txbody \
+ 	--signing-key-file txsign \
+ 	--network-magic … \
+ 	--tx-body-file txbody
+```
+
+You will need to give the correct Network Magic Id for the Testnet, as supplied by IOHK in the Genesis file (e.g. 42).
+
+5. Confirm that your node is synced, and then submit your transaction to the Blockchain:
+
+```bash
+$ cardano-cli shelley transaction submit \
+ 	--network-magic … \
+ 	--tx-body-file txbody
+```
+
+You should receive a confirmation that the transaction has been submitted to your local instance of the node, but it may not yet be recorded on the blockchain.  If the node is not running or it cannot be contacted, you will see an error.  Just kill/restart the node in this case and try again.
+
+6. After 2 minutes (possibly earlier), your Ada should be transferred to your new address.
+
+```bash
+ 	$ cardano-cli shelley query filtered-utxo \
+ 	 	--address … \
+ 	 	--network-magic …
+```
+
+7. Finally, build a transaction that sends a total of 1,000 Ada from *myaddr* to two different addresses (a multi-address transaction).
+
+```bash
+$ cardano-cli shelley address key-gen …
+$ cardano-cli shelley address key-gen …
+$ cardano-node shelley transaction build-raw …
+```
+
+The required fee will be higher than before, since part of the cost is based on the number of addresses that the output is sent to.  You can check that the fee will be sufficient before you build the transaction using:
+
+```bash
+$ cardano-node shelley transaction calculate-min-fee …
+```
+
+Sign, submit and wait for the transaction to be processed as before.
+
+```bash
+$ cardano-cli shelley transaction sign …
+$ cardano-node shelley transaction submit …
+```
+
+Once the transaction has completed, you will be able to verify that each of the addresses has received the correct amount of Ada.  Note that you now own four different addresses, each of which holds some Ada.  As you continue to process transactions and send/receive Ada to more addresses, your Ada will gradually become split among many different addresses, creating fragmentation.  This is not normally a big problem for the typical user who only owns a small amount of Ada, but can become a large problem for exchanges (as well as for whales, like you on the Testnet).
+
+8. Optional Exercise (Easy).
+
+Exchange some Ada with another Pioneer
+
+9. Optional Exercise (Easy).
+
+Defragment your Ada so that it is all held in one place.
```

Comment:
There is no documentation/tutorial explaining (de)fragmentation and it would be very useful to have one, especially considering that we bring this up into the exercise.

Reply:
I'm not sure a tutorial is the right thing here? (De-)fragmentation is more a general concept, like "utxo", "hash" etc., so it would rather belong into some sort of "blockchain glossary".